### PR TITLE
[TTAHUB-456] Add goal status visualization to frontend

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Added goal status visualization to frontend (bar graph showing status/counts for a given recipients goals). It can be found on the goals & objectives tab of the recipient record page.

## How to test
View the goal status visualization. Confirm that it shows the counts returned from the backend and that filters apply to it. Confirm that the data can be viewing as a simple HTML table rather than a horizontal bar graph.

## Issue(s)
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-456

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/widgets/GoalStatusGraph.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  count, percentage, label, color, ratio, total,
+}) {
+  const style = {
+    // 0/0 is NaN
+    width: Number.isNaN(percentage) ? '0%' : `${percentage * 100}%`,
+    backgroundColor: color,
+  };
+
+  const readablePercentage = `${(
+    percentage * 100).toLocaleString('en-us', { minimumFractionDigits: 0, maximumFractionDigits: 0 })
```
Comment (jasalisbury) [inline frontend/src/widgets/GoalStatusGraph.js:32]: I get "NaN percentage of...." with zero goals

File: frontend/src/widgets/GoalStatusGraph.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  );
+
+  useEffect(() => {
+    if (!data) {
+      return;
+    }
+
+    const newBars = GOAL_STATUSES.map((status, index) => ({
+      ratio: `${data[status]}/${data.total}`,
+      percentage: data[status] / data.total,
```
Comment (jasalisbury) [inline frontend/src/widgets/GoalStatusGraph.js:79]: Having a zero total making these values "NaN" is handled?

File: frontend/src/widgets/GoalStatusGraph.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from '@trussworks/react-uswds';
+import withWidgetData from './withWidgetData';
+import Container from '../components/Container';
+import AccessibleWidgetData from './AccessibleWidgetData';
+
+const GOAL_STATUSES = [
```
[the remark this answers is not in the document]
Comment (thewatermethod) [reply frontend/src/widgets/GoalStatusGraph.js:8]: Do you know what the process is going to be for giving all the goals that lack status a status?
On the frontend, we've been using the sentence casing. Do we need a ticket to change the database to match the frontend?

[the remark this answers is not in the document]
Comment (jasalisbury) [reply frontend/src/widgets/GoalStatusGraph.js:8]: They are going to get status through the table view's context menu (`set goal In progress`, or however it is worded). We can use sentence case, the statuses we have from smartsheet are "Not Started", "Ceased/Suspended", "In Progress", and "Closed", but we can write a migration to update (at the least we will need an update for "Ceased/Suspended" to update to "Ceased").

**Comment by jasalisbury** [comment]
Get an error adding a status filter with `not started`, `closed`, and `in progress`. This is for recipient 4100, `Adams and Sons` region 1, if you have the DB dump loaded.
Screenshot from 2022-02-11 14-45-55

**Comment by thewatermethod** [comment]
This should be good now

**Review by jasalisbury — APPROVED** [review]
:ship: I imagine there will be updates around the naming of `ceased`/`suspended` but otherwise looks good to me!

**Pull request merged**